tests/integ/modin/frame/test_assign.py

```diff
@@ -0,0 +1,163 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import re
+
+import modin.pandas as pd
+import pandas as native_pd
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
+from tests.integ.modin.utils import (
+    assert_snowpark_pandas_equals_to_pandas_without_dtypecheck,
+    create_test_dfs,
+    eval_snowpark_pandas_result,
+)
+
+
+@sql_count_checker(query_count=7, join_count=1)
+def test_assign_basic_series():
+    snow_df, native_df = create_test_dfs(
+        [[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+        columns=pd.Index(list("abc"), name="columns"),
+        index=pd.Index([0, 1, 2], name="index"),
+    )
+    native_df.columns.names = ["columns"]
+    native_df.index.names = ["index"]
+
+    def assign_func(df):
+        if isinstance(df, pd.DataFrame):
+            return df.assign(new_col=pd.Series([10, 11, 12]))
+        else:
+            return df.assign(new_col=native_pd.Series([10, 11, 12]))
+
+    eval_snowpark_pandas_result(snow_df, native_df, assign_func)
+
+
+@pytest.mark.parametrize("new_col_value", [2, [10, 11, 12], "x"])
+def test_assign_basic_non_pandas_object(new_col_value):
+    join_count = 2 if isinstance(new_col_value, list) else 0
+    with SqlCounter(query_count=7, join_count=join_count):
+        snow_df, native_df = create_test_dfs(
+            [[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+            columns=pd.Index(list("abc"), name="columns"),
+            index=pd.Index([0, 1, 2], name="index"),
+        )
+        native_df.columns.names = ["columns"]
+        native_df.index.names = ["index"]
+        eval_snowpark_pandas_result(
+            snow_df, native_df, lambda df: df.assign(new_column=new_col_value)
+        )
+
+
+@sql_count_checker(query_count=7, join_count=2)
+def test_assign_invalid_column_length_negative():
```

Comment:
This test illustrates one difference between us and vanilla pandas - since we maintain laziness, we don't check the length of the dataframe we are assigning to, so if the user attempts to assign something that is too short, we broadcast the last value, while pandas errors out. Should we document this anywhere?

Reply:
Yes, in the doc string.